CONTRIBUTING.md

```diff
@@ -0,0 +1,106 @@
+
+### The primary customers of the F# repository are users of the dotnet SDK, Visual Studio, Rider and Ionide. At all times their experience is paramount in our mind.
+
+  We are very accepting of community pull requests, there are however, a set of firm considerations that we hold to when reviewing PRs and determining what to merge.  These have been developed over the years to maintain the quality of the product and the experience that F# developers have when installing and upgrading the dotnet SDK and Visual Studio.
+
+- Does the change fix something that needs fixing, is there an issue, does the issue indicate a real problem?
+- Does the change improve the readability of something that needs improvement?
+- Does the change add a feature that is approved for adding?
+- Does the code match or improve of the existing codebase?
+- Is the performance improvement measured and can regressions be identified?
+- Will our existing customers be able to without effort upgrading the **Major** release of an SDK or VS?
+- Will our existing customers be able to without effort upgrading the **Minor** release of an SDK or VS?
+- Does it break VS, Rider or Ionide?
+- Does it have adequate testing?
+- Do all existing tests run unmodified?
+
+In general answers to the above should be **Yes**.  A **No** to any of them is not disqualifying of the PR, however a no answer will need an explanation and a discussion.
+ 
+There are additional considerations
+- Is the risk of accepting this change High or even Medium
+  For an acceptable PR with a high risk, we will definitely need to discuss mitigations for the risk.  A decision to upgrade the SDK or VS needs to be always low risk for our customers, they have businesses to run, they don't want to have to deal with our - risky behavior.  We may defer or delay risky PRs into a later release or abandon it.
+- Is the change as small as possible
+- Should it be chopped up into smaller chunks
+- Is the cost of reviewing the change worth the improvement made by the change
+Again, some PR’s are too big or provide too little value to merge.
+
+
+### Resources to help you get started
+
+Here are some resources to help you get started on how to contribute code or new content.
+
+- [Developers Guide](https://github.com/dotnet/fsharp/blob/main/DEVGUIDE.md) to get started on building the source code on your own.
+- [Test Guide](https://github.com/dotnet/fsharp/blob/main/TESTGUIDE.md) how to build run and work with test cases.
+- [F# compiler guide](https://github.com/dotnet/fsharp/blob/main/docs/index.md)
+- [F# language specification](https://fsharp.org/specs/language-spec/)
+- [F# language design](https://github.com/fsharp/fslang-design/)
+- [F# language suggestions](https://github.com/fsharp/fslang-suggestions/)
+- [help wanted](https://github.com/dotnet/fsharp/labels/help%20wanted) where to start
+
+### Submitting a pull request
```

Comment:
Maybe add a note about draft PRs - they can be fine for experimentation and give us a sign to not fully review them too early.

Reply:
WIP already exists as an option. --- so not really necessary to call it out here.